ci/prow/e2e-incluster-build

```diff
@@ -0,0 +1,70 @@
+##!/usr/bin/env bash
+
+set -euxo pipefail
+
+echo "Deploy KMMO..."
+make deploy
+
+echo "Wait until the KMMO Deployment is Available"
+echo "And Describe the Deployment / pods and get their YAML if that failed..."
+timeout 1m kubectl wait --for condition=Available deployments.apps -n kmm-operator-system kmm-operator-controller-manager || kubectl describe deployments.apps -n kmm-operator-system kmm-operator-controller-manager && kubectl get -o yaml deployments.apps -n kmm-operator-system kmm-operator-controller-manager && kubectl describe pod -n kmm-operator-system && kubectl get -o yaml pod -n kmm-operator-system
+
+echo "Create a build secret..."
+kubectl create secret generic build-secret --from-literal=ci-build-secret=super-secret-value
+
+      # The minikube registry-alias addon creates a Job that adds registry.minikube to the CoreDNS configuration.
+      # https://github.com/kubernetes/minikube/blob/master/deploy/addons/registry-aliases/patch-coredns-job.tmpl
+      # This job sometimes does not finish before the operator starts looking for the image in the registry, which
+      # results in failed DNS resolution.
+      # Add a job that tries to resolve registry.minikube, like the operator.
+      # Wait up to 6 minutes, which corresponds to our job's maximum lifetime including backoffs.
```

Comment:
```suggestion
# The minikube registry-alias addon creates a Job that adds registry.minikube to the CoreDNS configuration.
# https://github.com/kubernetes/minikube/blob/master/deploy/addons/registry-aliases/patch-coredns-job.tmpl
# This job sometimes does not finish before the operator starts looking for the image in the registry, which
# results in failed DNS resolution.
# Add a job that tries to resolve registry.minikube, like the operator.
# Wait up to 6 minutes, which corresponds to our job's maximum lifetime including backoffs.
```

Reply:
Thinking more about this, I think it rather belongs to the infra setup. We won't use the minikube internal registry midstream.

Reply:
We won't use minikube at all m/s :) 
+1 on moving to infra.